mm2src/adex_cli/src/scenarios/init_mm2_cfg.rs

```diff
@@ -0,0 +1,302 @@
+            spaces: false,
+            exclude_similar_characters: false,
+            strict: true,
+        };
+        pg.generate_one()
+            .map_err(|error| error!("Failed to generate password: {error}"))
+    }
+
+    fn inquire_allow_weak_password(&mut self) -> Result<(), ()> {
+        self.allow_weak_password = Confirm::new("Allow weak password:")
+                .with_default(false)
+                .with_placeholder("No")
+                .with_help_message(r#"If true, will allow low entropy rpc_password. If false rpc_password must not have 3 of the same characters in a row, must be between 8-32 characters in length, must contain at least one of each of the following: numeric, uppercase, lowercase, special character (e.g. !#$*). It also can not contain the word "password", or the chars <, >, and &. Defaults to false."#)
+                .prompt()
+                .map_err(|error| {
+                    error!("Failed to get allow_weak_password: {error}");
+                })?
+                .into();
+        Ok(())
+    }
+
+    fn inquire_userhome(&mut self) -> Result<(), ()> {
+        self.userhome = CustomType::<InquireOption<String>>::new("What is userhome:")
+                .with_placeholder(DEFAULT_OPTION_PLACEHOLDER)
+                .with_help_message(r#"The path to your home, called from your environment variables and entered as a regular expression. Example: /${HOME#"/"}"#)
+                .prompt()
+                .map_err(|error| {
+                    error!("Failed to get userhome: {error}");
+                })?.into();
+        Ok(())
+    }
+
+    fn inquire_rpcip(&mut self) -> Result<(), ()> {
+        self.rpcip = CustomType::<InquireOption<Ipv4Addr>>::new("What is rpcip:")
+            .with_placeholder(DEFAULT_OPTION_PLACEHOLDER)
+            .with_help_message("IP address to bind to for RPC server. Optional, defaults to 127.0.0.1")
+            .prompt()
+            .map_err(|error| {
+                error!("Failed to get rpcip: {error}");
+            })?
+            .into();
+        Ok(())
+    }
+
+    fn inquire_rpcport(&mut self) -> Result<(), ()> {
+        self.rpcport = CustomType::<InquireOption<u16>>::new("What is the rpcport:")
+            .with_help_message(r#"Port to use for RPC communication. Optional, defaults to 7783"#)
+            .with_placeholder(DEFAULT_OPTION_PLACEHOLDER)
+            .prompt()
+            .map_err(|error| {
+                error!("Failed to get rpcport: {error}");
+            })?
+            .into();
+        Ok(())
+    }
+
+    fn inquire_rpc_local_only(&mut self) -> Result<(), ()> {
+        self.rpc_local_only = CustomType::<InquireOption<bool>>::new("What is rpc_local_only:")
+                .with_parser(OPTION_BOOL_PARSER)
+                .with_formatter(DEFAULT_OPTION_BOOL_FORMATTER)
+                .with_default_value_formatter(DEFAULT_DEFAULT_OPTION_BOOL_FORMATTER)
+                .with_default(InquireOption::None)
```

Comment:
Default is `true`
https://github.com/KomodoPlatform/atomicDEX-API/blob/0058f7107509aade6268c56cf3f33933d8220d3b/mm2src/mm2_main/src/rpc.rs#L207
should we really have `rpc_local_only: Option<bool>,` in `Mm2Cfg`? can't it just be `bool`? Same for `i_am_seed` which defaults to `false`

Reply:
answered [before](https://github.com/KomodoPlatform/atomicDEX-API/pull/1729#discussion_r1148836064)